File: practice/algorithms/greedy/priyanka_and_toys/PriyankaAndToys.scala
object PriyankaAndToys extends App {
  val lines = Source.stdin.getLines().toList
  val n = Integer.parseInt(lines.head)
  val weights = lines.tail.head.split("\\s+").map(_.toInt).sorted
  var price = 1
  var lastWeight = weights(0)

  var i = 1
  while ( {
    i < weights.length
  }) {
    if (lastWeight + 4 >= weights(i)) {
      // lastWeight = toys[i];
    }
    else {
      price += 1
      lastWeight = weights(i)
    }

    {
      i += 1; i - 1
    }
  }
  println(price)
}

File: practice/algorithms/greedy/priyanka_and_toys/PriyankaAndToys.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const n = parseInt(lines[0]);
    const weights = lines[1].split(" ").map(i => parseInt(i)).sort((a, b) => a - b);

    let price = 1;
    let lastWeight = weights[0];
    for (let i = 1; i < weights.length; i++)
    {
        if(lastWeight + 4 >= weights[i])
        {
            // lastWeight = toys[i];
        }
        else
        {
            price++;
            lastWeight = weights[i];
        }
    }
    console.log(price);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));